pwr.md

```diff
@@ -0,0 +1,144 @@
+Configuration is done by first creating core and/or CPU object lists, these objects will store the characteristics of the physical processor.
+Modifying those objects and committing the changes enables the module to take the necessary steps to make those changes on the CPU hardware.
+
+## Initialization 
+Creation of the cpu and core object lists is done using the `get_cores()/get_cpus()` functions, which return a list of the respective objects.
+
+```python 
+import pwr
+
+cores = pwr.get_cores()
+cpus = pwr.get_cpus()
+```
+
+## Adjusting Power Configuration
+### Modifying
+Each cpu and core object have attributes which replicate the capabilities and stats of the physical cores and cpus.
+### cpu
+* `cpu_id`                  # CPU id number
+* `physical_id`             # physical CPU number
+* `core_list`               # list of core objects on this CPU
+* `sst_bf_enabled`          # base frequency enabled in BIOS
+* `sst_bf_configured`       # core configured to use base frequencies
+* `turbo_enabled`           # turbo enabled flag
+* `hwp_enabled`             # HWP enabled flag
+* `base_freq`               # base frequency
+* `all_core_turbo_freq`     # all core turbo frequency
+* `highest_freq`            # highest available frequency
+* `lowest_freq`             # lowest active frequency
+* `uncore_hw_max`           # max available uncore frequency
+* `uncore_hw_min`           # min available uncore frequency
+* `uncore_freq`             # current uncore frequency
+* `uncore_max_freq`         # max desired uncore frequency
+* `uncore_min_freq`         # min desired uncore frequency
+
+### core
+* `core_id`                 # core id number
+* `cpu`                     # this cores cpu object
+* `thread_siblings`         # list of thread siblings
+* `high_prioirty`           # high/normal priority
+* `base_freq`               # base frequency [2300Mhz]
+* `sst_bf_base_freq`        # priority based frequency [2100Mhz/2700Mhz]
+* `all_core_turbo_freq`     # all core turbo frequency [2800Mhz]
+* `highest_freq`            # highest frequency available [3900Mhz]
+* `lowest_freq`             # lowest active frequency [800Mhz]
+* `curr_freq`               # current core frequency
+* `min_freq`                # desired low frequency
+* `max_freq`                # desired high frequency
+* `epp`                     # energy performance preference
+
+The **core** object attributes which can be adjusted are `max_freq`, `min_freq` and `epp`, min and max set the desired core frequency at which an application wants the core to operate within. 
+>Available frequencies are between `lowest_freq` and `highest_freq`, with some defined freqeuncies being `base_freq`/`sst_bf_base_freq`/`all_core_turbo_freq` if configuration allows.
+
+`epp` sets the desired performance profile for the core.
+>Available profiles are `default`, `performance`, `balance_performance`, `balance_power` and `power`.
```

Comment:
Available profiles depend on the kernel. Do we provide an externally visible list of these? If not, then i suppose documenting these here is OK as it's consistent with what i've seen so far on all the SKU's.